app/vmalert/manager.go

```diff
@@ -97,19 +98,33 @@ func (m *manager) update(ctx context.Context, path []string, validateTpl, valida
 	for _, og := range m.groups {
 		ng, ok := groupsRegistry[og.ID()]
 		if !ok {
+			startAt := time.Now()
 			// old group is not present in new list
 			// and must be stopped and deleted
 			og.close()
 			delete(m.groups, og.ID())
+			logger.Infof("group %q delete success, took %v seconds", og.Name, time.Since(startAt).Seconds())
 			og = nil
 			continue
 		}
-		og.updateCh <- ng
+		go func(og *Group, ng *Group) {
+			groupReloadRequested.Inc()
+			select {
+			case <-ctx.Done():
+				return
+			case <-og.doneCh:
+				return
+			case og.updateCh <- ng:
+				return
+			}
+		}(og, ng)
 		delete(groupsRegistry, ng.ID())
 	}
 
 	for _, ng := range groupsRegistry {
+		startAt := time.Now()
 		m.startGroup(ctx, ng, restore)
+		logger.Infof("group %q new start success, took %v seconds", ng.Name, time.Since(startAt).Seconds())
```

Comment:
pls rename `new start success` to `started`. And leave only one "start" message to avoid double logging:
```
2020-08-16T08:32:31.505Z        info    app/vmalert/manager.go:127      group "TestGroup2" new start success, took 6.182e-06 seconds
2020-08-16T08:32:31.505Z        info    app/vmalert/group.go:188        group "TestGroup2" started; interval=2s; concurrency=2
```